### PR TITLE
feat(j-s): Robot mail for received indictments

### PR DESCRIPTION
**Pull request by unakb**
[Roboti - ákærur (dagsetning, tegund og nöfn)](https://app.asana.com/0/1199153462262248/1207400397825206)

## What

Added new message for received indictment cases to be sent to the court robot

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced a new method to handle updating indictment case information in the judicial system.

- **Bug Fixes**
  - Renamed query parameter from 'lang' to 'locale' to better reflect its purpose and added locale validation.

- **Tests**
  - Added tests to verify guards related to delivering indictment information to court.

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts [hunk 1]
```diff
@@ -771,7 +818,7 @@
       nowFactory() // The API requires a date so we send now as a dummy date
 
     return this.policeService.updatePoliceCase(
-      user,
+    user,
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts]: This is unrelated to this PR

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 2]
```diff
@@ -602,20 +602,28 @@ export class CaseService {
...
+        })
+        theCase.policeCaseNumbers.forEach((policeCaseNumber) =>
+          messages.push({
+            type: MessageType.DELIVERY_TO_POLICE_CASE_FILES_RECORD,
+            user,
+            caseId: theCase.id,
+            elementId: policeCaseNumber,
+          }),
+        )
+      }
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/backend/src/app/modules/case/case.service.ts:626]: The code is pretty self explanatory, this comment would be very repetitive.

File: apps/judicial-system/backend/src/app/modules/court/court.service.ts [hunk 3]
```diff
@@ -555,6 +556,53 @@ export class CourtService {
...
+          courtCaseNumber,
+          receivedByCourtDate,
+          indictmentDate,
+          policeCaseNumber,
+        },
+        error,
+      )
+
+      throw error
+    }
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/backend/src/app/modules/court/court.service.ts:603]: This is in line with what we are doing today so I will keep it this way until we decide to refactor.

File: apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverIndictmentInfoToCourtGuards.spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { CanActivate } from '@nestjs/common'
+
+import { indictmentCases } from '@island.is/judicial-system/types'
+
+import { CaseCompletedGuard } from '../../guards/caseCompleted.guard'
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverIndictmentInfoToCourtGuards.spec.ts:5]: Please remove unused import.

**Review by gudjong — APPROVED** [review]
Minor comment

**Pull request merged**